src/workerd/io/io-context.h

```diff
@@ -1392,19 +1392,6 @@ kj::_::ReducePromises<RemoveIoOwn<T>> IoContext::awaitJs(jsg::Lock& js, jsg::Pro
         fulfiller->reject(JSG_KJ_EXCEPTION(FAILED, Error, "Promise will never complete."));
       }
     }
-
-    kj::Maybe<kj::StringPtr> finalize() override {
-      if (!isDone) {
-        reject();
-        isDone = true;
-        return "A hanging Promise was canceled. This happens when the worker runtime is waiting "
```

Comment:
I wonder if there's a way to have this error still exist?

Reply:
This message was only logged to the devtools console (only in preview). From what I can tell, most people hit by this error never saw this extended description. I think what we really need to do is extend the error in the exception we actually throw. I do want to do that but I kind of want to do it in a separate PR since I think a bunch of people will be interested in bikeshedding it.